foreshadow/transformers/base.py

```diff
@@ -440,91 +483,56 @@ def set_params(self, **params):
+    @abstractmethod
+    def pick_transformer(self, X, y=None, **fit_params):
+        """Pick the correct transformer object for implementations.
 
         Args:
             X (:obj:`pandas.DataFrame`): Input X data
             y (:obj: 'pandas.DataFrame'): labels Y for data
             **fit_params (dict): Parameters to apply to transformers when
                 fitting
 
-        Raises:
-            NotImplementedError: need to override
-
         """
-        raise NotImplementedError(
-            "WrappedTransformer _get_transformer was not implimented."
-        )
+        pass
 
-    def _verify_transformer(self, X, y=None, refit=False, **fit_params):
+    def resolve(self, X, y=None, **fit_params):
         """Verify transformers have the necessary methods and attributes.
 
         Args:
             X: input observations
             y: input labels
-            refit: ignore if previous fit was completed and refit
             **fit_params: params to fit
 
-        Raises:
-            AttributeError: if invalid wrapped transformer
-
         """
-        # If refit transformer needs to be re-resolved
-        if refit:
-            self._transformer = None
-
-        # If nothing needs to be done go ahead and return
-        if self._transformer is not None:
-            return
-
-        # If overriding, resolve the override and create the object
-        if self.override is not None:
-            self._transformer = self._resolve(self.override)(**self.kwargs)
-        # If not use _get_transformer to get the object
-        else:
-            self._transformer = self._get_transformer(X, y, **fit_params)
-
-        # Check attributes
-        tf = getattr(self._transformer, "transform", None)
-        fittf = getattr(self._transformer, "fit_transform", None)
-        fit = getattr(self._transformer, "fit", None)
-
-        nm = hasattr(self._transformer, "name")
-        keep = hasattr(self._transformer, "keep_columns")
-
-        pipe = hasattr(self._transformer, "steps")
-        parallel = hasattr(self._transformer, "transformer_list")
-
-        # Check callable status of methods
-        if not (
-            callable(tf)
-            and callable(fittf)
-            and callable(fit)
-            and (nm and keep)
-            or pipe
-            or parallel
-        ):
-            raise AttributeError(
-                "Invalid WrappedTransformer. Get transformer returns invalid "
-                "object"
-            )
-
-        # Propagate name and keep_columns attributes to transformer
-        self._transformer.name = self.name
-        self._transformer.keep_columns = self.keep_columns
+        # If override is passed in or set, all types of resolves are turned
+        # off.
+        # Otherwise, force_reresolve will always resolve on each fit.
+
+        # If force_reresolve is set, always re-resolve
+        if self.force_reresolve:
+            self.should_resolve = True
```

Comment:
why do we need should_resolve? it seems as though it should be the standard behaviour.

Reply:
Will fix. will have singular statement checking whether force resolve is set, and will update should resolve as a result of that.